op-batcher/batcher/channel_manager_test.go

```diff
@@ -668,3 +668,24 @@ func TestChannelManager_Requeue(t *testing.T) {
 
 	require.NotContains(t, m.blocks, blockA)
 }
+func TestChannelManager_ChannelOutFactory(t *testing.T) {
+	type ChannelOutWrapper struct {
+		derive.ChannelOut
+	}
+
+	l := testlog.Logger(t, log.LevelCrit)
+	cfg := channelManagerTestConfig(100, derive.SingularBatchType)
+	m := NewChannelManager(l, metrics.NoopMetrics, cfg, defaultTestRollupConfig)
+	m.SetChannelOutFactory(func(cfg ChannelConfig, rollupCfg *rollup.Config) (derive.ChannelOut, error) {
+		co, err := NewChannelOut(cfg, rollupCfg)
+		if err != nil {
+			return nil, err
+		}
+		return &ChannelOutWrapper{
+			ChannelOut: co,
+		}, nil
```

Comment:
Perhaps add a comment here about this wrapper allowing for custom behaviour (otherwise seems weird in isolation to not just` return NewChannelOut(cfg, rollupConfig)`.

Reply:
good idea, done